data-pipeline-ffi/src/trace_exporter.rs

```diff
@@ -111,8 +111,11 @@ pub unsafe extern "C" fn ddog_trace_exporter_send(
     trace_count: usize,
 ) -> MaybeError {
     // TODO - handle errors - https://datadoghq.atlassian.net/browse/APMSP-1095
+    let static_trace: ByteSlice<'static> = std::mem::transmute(trace);
+    let tinybytes_trace = tinybytes::Bytes::from(static_trace);
+
```

Comment:
I don't know what would be more idiomatic using transmute to modify the lifetime or wrapping the pointer in as struct with a lifetime marker as 
in:
```
let data = ByteSlice::from_raw_parts(trace.as_mut_ptr(), trace.len());
let tinybytes_trace = tinybytes::Bytes::from(data);
```